src/plugin/featureaction/featuresoundaction.js

```diff
@@ -0,0 +1,164 @@
+goog.provide('plugin.im.action.feature.SoundAction');
+
+goog.require('os.feature');
+goog.require('os.im.action.AbstractImportAction');
+goog.require('os.implements');
+goog.require('os.legend');
+goog.require('os.legend.ILegendRenderer');
+goog.require('os.object');
+goog.require('os.xml');
+
+/**
+ * Tag names used for XML persistence.
+ * @enum {string}
+ */
+plugin.im.action.feature.SoundActionTagName = {
+  SOUND: 'sound'
+};
+
+/**
+ * Import action that sets the sound for a {@link ol.Feature}.
+ * @extends {os.im.action.AbstractImportAction<ol.Feature>}
+ * @implements {os.legend.ILegendRenderer}
+ * @constructor
+ */
+plugin.im.action.feature.SoundAction = function() {
+  plugin.im.action.feature.SoundAction.base(this, 'constructor');
+
+  this.id = plugin.im.action.feature.SoundAction.ID;
+  this.label = plugin.im.action.feature.SoundAction.LABEL;
+  this.xmlType = plugin.im.action.feature.SoundAction.ID;
+  this.configUI = plugin.im.action.feature.SoundAction.CONFIG_UI;
+
+
+  /**
+   * Unique identifier for this action.
+   * @type {number}
+   * @protected
+   */
+  this.uid = goog.getUid(this);
+
+  /**
+   * The feature sound config.
+   * @type {!Object}
+   */
+  this.soundConfig = /** @type {!Object} */ (os.object.unsafeClone(
+      plugin.im.action.feature.SoundAction.DEFAULT_CONFIG));
+};
+goog.inherits(plugin.im.action.feature.SoundAction,
+    os.im.action.AbstractImportAction);
+
+os.implements(plugin.im.action.feature.SoundAction,
+    os.legend.ILegendRenderer.ID);
+
+/**
+ * Action identifier.
+ * @type {string}
+ * @const
+ */
+plugin.im.action.feature.SoundAction.ID = 'featureSoundAction';
+
+/**
+ * Property set on features to indicate they're using a feature sound action.
+ * @type {string}
+ * @const
+ */
+plugin.im.action.feature.SoundAction.FEATURE_ID = '_featureSoundAction';
+
+/**
+ * Action Label.
+ * @type {string}
+ * @const
+ */
+plugin.im.action.feature.SoundAction.LABEL = 'Set Sound';
+
+/**
+ * Action edit UI.
+ * @type {string}
+ * @const
+ */
+plugin.im.action.feature.SoundAction.CONFIG_UI = 'featureactionsoundconfig';
+
+/**
+ * The default sound configuration.
+ * @type {!Object}
+ * @const
+ */
+plugin.im.action.feature.SoundAction.DEFAULT_CONFIG = {
+  'sound': 'Default'
+
+};
+
+/**
+ * @inheritDoc
+ */
+plugin.im.action.feature.SoundAction.prototype.execute = function(items) {
+  for (var i = 0; i < items.length; i++) {
```

Comment:
I would recommend removing the for loop and calling `play` once. You don't want the sound playing multiple times if several matching features are loaded.

I would also suggest giving the user a way to set the "time between plays" parameter to `play`. Default it to something infrequent (we have used 30 seconds in similar UI's), and enforce a reasonable minimum (probably 1 or 5 seconds) to prevent the sound from playing too often.